FlowCrypt/Controllers/Setup/SetupGenerateKeyViewController.swift

```diff
@@ -119,33 +115,35 @@ private actor Service {
             try appContext.passPhraseService.savePassPhrase(with: passPhrase, storageMethod: .memory)
         }
 
-        await submitKeyToAttesterAndShowAlertOnFailure(
-            email: userId.email,
-            publicKey: encryptedPrv.key.public,
-            viewController: viewController
-        )
-
         // sending welcome email is not crucial, so we don't handle errors
         _ = try? await attester.testWelcome(
             email: userId.email,
             pubkey: encryptedPrv.key.public
         )
     }
 
-    private func submitKeyToAttesterAndShowAlertOnFailure(
+    @MainActor
+    private func putKeypairs(encryptedPrv: CoreRes.GenerateKey, storageMethod: StorageMethod, passPhrase: String) throws {
+        try appContext.encryptedStorage.putKeypairs(
+            keyDetails: [encryptedPrv.key],
+            passPhrase: storageMethod == .persistent ? passPhrase: nil,
+            source: .generated,
+            for: user.email
+        )
+    }
+
+    private func submitKeyToAttester(
         email: String,
-        publicKey: String,
-        viewController: ViewController
-    ) async {
+        publicKey: String
+    ) async throws {
         do {
             _ = try await attester.update(
                 email: email,
                 pubkey: publicKey,
                 token: appContext.dataService.token
             )
         } catch {
-            let message = "Failed to submit Public Key"
-            await viewController.showAlert(error: error, message: message)
+            throw CreateKeyError.submitKey(error)
         }
     }
```

Comment:
This is consistent with our general error handling approach, but now you can't implement my request `we have to add a modal with retry/skip option` easily. For that, you would need local handling of the error like before.

Nevermind, since IMAP is a fringe usecase, we can leave it as is for now for simplicity.